src/hnsw/scan.c

```diff
@@ -211,6 +213,31 @@ bool ldb_amgettuple(IndexScanDesc scan, ScanDirection dir)
         if(value != scan->orderByData->sk_argument) pfree(DatumGetPointer(value));
     }
 
+    if(scanstate->current == scanstate->count) {
+        int             num_returned;
+        Datum           value;
+        Vector         *vec;
+        usearch_error_t error = NULL;
+
+        value = scan->orderByData->sk_argument;
+
+        vec = DatumGetVector(value);
+
+        /* double k and reallocate arrays to account for increased size */
+        k = k * 2;
```

Comment:
So you are using the static `k` to know how many items to retrieve next.
Could this be `k = scanstate->count * 2`?

It is very strange to see static variables in this context. I am pretty sure statics will do the wrong thing if two scans happen at the same time. So, if possible, let's get rid of the static variable.

Reply:
good catch, this was poorly thought out